### PR TITLE
fix: add timestamping cert chain revocation check during signing

### PR DESCRIPTION
**Pull request by Two-Hearts**
This PR adds tsa cert chain revocation check after signing to fail fast.

### CODE REVIEW

File: internal/revocation/revocation.go
Comment (JeyJeyGao) [inline internal/revocation/revocation.go:1]: We will have both `internal/revocation` and `internal/crl` folders. Is it better to reorganize the folders and move the `crl` package to `internal/revocation/crl`?

Comment (Two-Hearts) [reply internal/revocation/revocation.go:1]: Good idea!

**Review by JeyJeyGao — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by rgnote — APPROVED** [review]
LGTM

**Pull request merged**